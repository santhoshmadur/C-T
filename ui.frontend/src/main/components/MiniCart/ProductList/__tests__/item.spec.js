/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
/* eslint-disable react/display-name, @typescript-eslint/no-empty-function */
import React from 'react';
import { useItem } from '@magento/peregrine/lib/talons/MiniCart/useItem';

import Item from '../item';
import render from '../../../utils/test-utils';

jest.mock('@magento/venia-ui/lib/classify');
jest.mock('@magento/peregrine/lib/talons/MiniCart/useItem', () => ({
    useItem: jest.fn().mockReturnValue({
        isDeleting: false,
        removeItem: () => {}
    })
}));
jest.mock('react-router-dom', () => ({
    Link: ({ children, ...rest }) => <div {...rest}>{children}</div>
}));
// jest.mock('@magento/peregrine/lib/util/makeUrl', () => jest.fn(src => src));

const props = {
    product: {
        name: 'P1',
        url_key: 'product',
        url_suffix: '.html',
        thumbnail: {
            url: 'www.venia.com/p1'
        },
        variants: [
            {
                attributes: [
                    {
                        uid: 'Y29uZmlndXJhYmxlLzIyLzI='
                    }
                ],
                product: {
                    thumbnail: {
                        url: 'www.venia.com/p1-variant1'
                    }
                }
            },
            {
                attributes: [
                    {
                        uid: 'Y29uZmlndXJhYmxlLzIyLzM='
                    }
                ],
                product: {
                    thumbnail: {
                        url: 'www.venia.com/p1-variant2'
                    }
                }
            }
        ]
    },
    id: 'p1',
    quantity: 10,
    configurable_options: [
        {
            option_label: 'Color',
            value_label: 'red',
            id: 22,
            value_id: 2
        }
    ],
    handleRemoveItem: () => {},
    prices: {
        price: {
            value: 420,
            currency: 'USD'
        }
    }
};

test('Should render correctly', () => {
    const tree = render(<Item {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Should render correctly with out of stock product', () => {
    const outOfStockProps = {
        ...props,
        product: {
            ...props.product,
            stock_status: 'OUT_OF_STOCK'
        }
    };
    const tree = render(<Item {...outOfStockProps} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Should render correctly when configured to use variant thumbnail', () => {
    const variantThumbnailProps = {
        ...props,
        configurableThumbnailSource: 'itself'
    };
    const tree = render(<Item {...variantThumbnailProps} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Should disable delete icon while loading', () => {
    useItem.mockReturnValueOnce({
        isDeleting: true,
        removeItem: () => {}
    });
    const tree = render(<Item {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
});
