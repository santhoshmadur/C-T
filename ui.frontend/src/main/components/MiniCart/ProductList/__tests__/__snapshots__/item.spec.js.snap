// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should disable delete icon while loading 1`] = `
<div
  className="root_disabled"
>
  <div
    className="thumbnailContainer"
  >
    <div
      className="root container"
    >
      <img
        alt="P1"
        className="image placeholder"
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        width={100}
      />
      <img
        alt="P1"
        className="image notLoaded"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100px"
        src="https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=100&height=125&fit=cover"
        srcSet="https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
        width={100}
      />
    </div>
  </div>
  <div
    className="name"
  >
    P1
  </div>
  <dl
    className="options"
  >
    <div
      className="optionLabel"
    >
      <dt
        className="optionName"
      >
        Color :
      </dt>
      <dd
        className="optionValue"
      >
        red
      </dd>
    </div>
  </dl>
  <span
    className="quantity"
  >
    Qty : 10
  </span>
  <span
    className="price"
  >
    <span>
      $
    </span>
    <span>
      420
    </span>
    <span>
      .
    </span>
    <span>
      00
    </span>
     ea.
  </span>
  <span
    className="stockStatus"
  >
    
  </span>
  <button
    className="deleteButton"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="root"
    >
      <svg
        className="icon"
        fill="none"
        height={16}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="thumbnailContainer"
  >
    <div
      className="root container"
    >
      <img
        alt="P1"
        className="image placeholder"
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        width={100}
      />
      <img
        alt="P1"
        className="image notLoaded"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100px"
        src="https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=100&height=125&fit=cover"
        srcSet="https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
        width={100}
      />
    </div>
  </div>
  <div
    className="name"
  >
    P1
  </div>
  <dl
    className="options"
  >
    <div
      className="optionLabel"
    >
      <dt
        className="optionName"
      >
        Color :
      </dt>
      <dd
        className="optionValue"
      >
        red
      </dd>
    </div>
  </dl>
  <span
    className="quantity"
  >
    Qty : 10
  </span>
  <span
    className="price"
  >
    <span>
      $
    </span>
    <span>
      420
    </span>
    <span>
      .
    </span>
    <span>
      00
    </span>
     ea.
  </span>
  <span
    className="stockStatus"
  >
    
  </span>
  <button
    className="deleteButton"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="root"
    >
      <svg
        className="icon"
        fill="none"
        height={16}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Should render correctly when configured to use variant thumbnail 1`] = `
<div
  className="root"
>
  <div
    className="thumbnailContainer"
  >
    <div
      className="root container"
    >
      <img
        alt="P1"
        className="image placeholder"
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        width={100}
      />
      <img
        alt="P1"
        className="image notLoaded"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100px"
        src="https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=100&height=125&fit=cover"
        srcSet="https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
https://backend.test/media/catalog/product/www.venia.com/p1-variant1?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
        width={100}
      />
    </div>
  </div>
  <div
    className="name"
  >
    P1
  </div>
  <dl
    className="options"
  >
    <div
      className="optionLabel"
    >
      <dt
        className="optionName"
      >
        Color :
      </dt>
      <dd
        className="optionValue"
      >
        red
      </dd>
    </div>
  </dl>
  <span
    className="quantity"
  >
    Qty : 10
  </span>
  <span
    className="price"
  >
    <span>
      $
    </span>
    <span>
      420
    </span>
    <span>
      .
    </span>
    <span>
      00
    </span>
     ea.
  </span>
  <span
    className="stockStatus"
  >
    
  </span>
  <button
    className="deleteButton"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="root"
    >
      <svg
        className="icon"
        fill="none"
        height={16}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Should render correctly with out of stock product 1`] = `
<div
  className="root"
>
  <div
    className="thumbnailContainer"
  >
    <div
      className="root container"
    >
      <img
        alt="P1"
        className="image placeholder"
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        width={100}
      />
      <img
        alt="P1"
        className="image notLoaded"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100px"
        src="https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=100&height=125&fit=cover"
        srcSet="https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
https://backend.test/media/catalog/product/www.venia.com/p1?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
        width={100}
      />
    </div>
  </div>
  <div
    className="name"
  >
    P1
  </div>
  <dl
    className="options"
  >
    <div
      className="optionLabel"
    >
      <dt
        className="optionName"
      >
        Color :
      </dt>
      <dd
        className="optionValue"
      >
        red
      </dd>
    </div>
  </dl>
  <span
    className="quantity"
  >
    Qty : 10
  </span>
  <span
    className="price"
  >
    <span>
      $
    </span>
    <span>
      420
    </span>
    <span>
      .
    </span>
    <span>
      00
    </span>
     ea.
  </span>
  <span
    className="stockStatus"
  >
    Out-of-stock
  </span>
  <button
    className="deleteButton"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="root"
    >
      <svg
        className="icon"
        fill="none"
        height={16}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </span>
  </button>
</div>
`;
